clientele/framework/config.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Any
+
+import httpx
+
+
+@dataclass
```

Comment:
We should not use dataclass here. In fact this entire file should just be a copy and paste of the `Config` object we generate for clients.